### ISSUE TITLE
include way tags in instructions

### ISSUE THREAD

**Issue by emiltin** [Project-OSRM/osrm-backend#1342]
in our "I Bike CPH" bike router app for ios and android, we provide GPS turn-by-turn navigation. we would like to use things like way type, surface, etc. to provide additional cues. this is relevant for bikes, since many bike paths don't have names.

a workaround is to encode the tags as part of the way name in the lua profile, for example "Vesterbrogade {'highway':'primary'}" or "{'highway':'cycleway','surface':'gravel'}. but this takes up extra space. 

a more efficient way of storing tags and passing them to the front-end would be great. one idea would be to store the encoded tags as a separate way 'name', ie store "Vesterbrogade" and "{'highway':'primary'}" separately. since many ways would have the same tags, it would take up less space.

**Comment by emiltin** [comment]
related to #282.

**Comment by TheMarex** [comment]
What you basically want is a a mini key-value storage engine inside OSRM that maps edge/node ids to properties. Could be worthwhile to invest some time in this, as this often comes up. However implementing this is not trivial.
Should be doable without any runtime/memory penalty *if you don't use it*.

**Comment by emiltin** [comment]
right. preferably, the lua script will be able set the keys/values for each way.

**Comment by systemed** [comment]
I use the way name workaround at cycle.travel. The results are heavily tokenised (one character for surface type, etc.) which saves some space and download time. But yes, a PBF-like key-val store wouldn't be a bad thing.

**Comment by TheMarex** [comment]
This would also unblock https://github.com/Project-OSRM/osrm-backend/issues/77 since we could move the speed inside that field.

**Comment by marcioaguiar** [comment]
I'm also interested in this feature, specifically on the surface property. Maybe it could be customized on the profile?

**Comment by danpat** [comment]
If you can handle a second network round-trip, you can use this to get tag data:

https://github.com/mapbox/route-annotator
